llama-index-core/llama_index/core/callbacks/base.py

```diff
@@ -68,6 +68,15 @@ def __init__(self, handlers: Optional[List[BaseCallbackHandler]] = None):
                     )
             handlers.append(new_handler)
 
+        # if we passed in no handlers, use the global default
+        if len(handlers) == 0:
+            from llama_index.core.settings import Settings
+
+            # hidden var access to prevent recursion in getter
+            cb_manager = Settings._callback_manager
+            if cb_manager is not None:
+                handlers = cb_manager.handlers

```

Comment:
is the default value of Settings.callback_manager.handlers an empty list? if not, should we do that else change might break some assumptions downstream of people relying on by default this value to be []

Reply:
The default is an empty list yes